platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/AttributionDialogManager.java

```diff
@@ -141,18 +144,57 @@ public void onClick(@NonNull DialogInterface dialog, int which) {
   private void showMapFeedbackWebPage(int which) {
     Attribution[] attributions = attributionSet.toArray(new Attribution[attributionSet.size()]);
     String url = attributions[which].getUrl();
-    if (url.contains(MAP_FEEDBACK_URL)) {
-      url = buildMapFeedbackMapUrl(mapboxMap.getCameraPosition());
-    }
+    Logger.d(TAG, "attribution url: " + url);
+    url = buildMapFeedbackMapUrl();
+    Logger.d(TAG, "final feedback url: " + url);
     showWebPage(url);
   }
 
   @NonNull
-  private String buildMapFeedbackMapUrl(@Nullable CameraPosition cameraPosition) {
-    // appends current location to the map feedback url if available
-    return cameraPosition != null ? String.format(Locale.getDefault(),
-      MAP_FEEDBACK_LOCATION_FORMAT, cameraPosition.target.getLongitude(), cameraPosition.target.getLatitude(),
-      (int) cameraPosition.zoom) : MAP_FEEDBACK_URL;
+  private String buildMapFeedbackMapUrl() {
+    // TODO Add Android Maps SDK version to the query parameter, currently the version API is not available.
+
+    Uri.Builder builder = Uri.parse(MAP_FEEDBACK_URL).buildUpon();
+
+    CameraPosition cameraPosition = mapboxMap.getCameraPosition();
+    if (cameraPosition != null) {
+      builder.encodedFragment(String.format(Locale.getDefault(), MAP_FEEDBACK_URL_LOCATION_FRAGMENT_FORMAT,
+              cameraPosition.target.getLongitude(), cameraPosition.target.getLatitude(),
+              cameraPosition.zoom, cameraPosition.bearing, (int) cameraPosition.tilt));
+    }
+
+    String packageName = context.getPackageName();
+    if (packageName != null) {
+      builder.appendQueryParameter("referrer", packageName);
+    }
+
+    String accessToken = Mapbox.getAccessToken();
+    if (accessToken != null) {
+      builder.appendQueryParameter("access_token", accessToken);
+    }
+
+    Style style = mapboxMap.getStyle();
```

Comment:
You should be able to do the following, which allows you to get rid of the null check

```suggestion
mapboxMap.getStyle(new Style.OnStyleLoaded() {
      @Override
      public void onStyleLoaded(@NonNull Style style) {
      
      String styleUri = style.getUri();
      Logger.d(TAG, "style uri:" + styleUri);

      Pattern pattern = Pattern.compile(MAP_FEEDBACK_STYLE_URI_REGEX);
      Matcher matcher = pattern.matcher(styleUri);

      if (matcher.find()) {
        String styleApi = matcher.group(1);
        String styleOwner = matcher.group(2);
        String styleId = matcher.group(3);
        Logger.d(TAG, "styles API: " + styleApi);
        Logger.d(TAG, "styles owner: " + styleOwner);
        Logger.d(TAG, "styles id: " + styleId);

        builder.appendQueryParameter("owner", styleOwner)
                .appendQueryParameter("id", styleId);
      }
      }
    });
```

Reply:
@langsmith thanks for the comment! The reason I used getStyle() directly instead of callback is that we can not guarantee if or when the style is loaded, and user should be able to open the feedback webpage reasonably fast and send the feedback even if the style is not present(for eample, camera position is a more important parameter for the feedback service). 
With the above thoughts, I didn't use the async callback method :)

Reply:
Got it, ok.